pkg/csi/service/common/vsphereutil.go

```diff
@@ -773,6 +783,11 @@ func QueryAllVolumeSnapshots(ctx context.Context, volManager cnsvolume.Manager,
 		return nil, "", err
 	}
 	for _, queryResult := range queryResultEntries {
+		if queryResult.Error != nil {
+			log.Warnf("unexpected fault %+v received in QuerySnapshots result: %+v, ignoring..",
+				queryResult.Error.Fault, queryResult)
+			continue
```

Comment:
Same question here.

Reply:
same reasoning as above, we are re-iterating the same results.